RxSwift/Observables/Implementations/Timeout.swift

```diff
@@ -0,0 +1,122 @@
+//
+//  Timeout.swift
+//  Rx
+//
+//  Created by Tomi Koskinen on 13/11/15.
+//  Copyright © 2015 Krunoslav Zaher. All rights reserved.
+//
+
+import Foundation
+
+class TimeoutSink<ElementType, Scheduler: SchedulerType, O: ObserverType where O.E == ElementType>: Sink<O>, LockOwnerType, ObserverType {
+    typealias E = ElementType
+    typealias Parent = Timeout<E, Scheduler>
+    
+    private let _parent: Parent
+    
+    let _lock = NSRecursiveLock()
+
+    private let _timerD = SerialDisposable()
+    private let _subscription = SerialDisposable()
+    
+    private var _id = 0
+    private var _switched = false
+    
+    init(parent: Parent, observer: O) {
+        _parent = parent
+        super.init(observer: observer)
+    }
+    
+    func run() -> Disposable {
+        let original = SingleAssignmentDisposable()
+        _subscription.disposable = original
+        
+        _createTimeoutTimer()
+        
+        original.disposable = _parent._source.subscribeSafe(self)
+        
+        return StableCompositeDisposable.create(_subscription, _timerD)
+    }
+
+    func on(event: Event<E>) {
+        switch event {
+        case .Next:
+            var onNextWins = false
+            
+            _lock.performLocked() {
+                onNextWins = !self._switched
+                if onNextWins {
+                    self._id++
```

Comment:
I think we should use `self._id = self._id &+ 1`, otherwise these bad boys could be party crashers :)